servant-server/src/Servant/Server/Internal.hs

```diff
@@ -276,6 +286,42 @@ instance OVERLAPPING_
     where method = reflectMethod (Proxy :: Proxy method)
           status = toEnum . fromInteger $ natVal (Proxy :: Proxy status)
 
+
+instance ( MimeRender ctype a, ReflectMethod method, Framing framing ctype
+         ) => HasServer (Stream method framing ctype a) context where
+
+  type ServerT (Stream method framing ctype a) m = m (StreamGenerator a)
```

Comment:
I think we should have something more low-level as a base primitive, maybe as low as a conversion from `a` to [`StreamingBody`](http://hackage.haskell.org/package/wai-3.2.1.1/docs/Network-Wai.html#t:StreamingBody)

You will be able to build `Framed` on top of it.

Reply:
Could you give a use case?